tests/integration/test_integration_all_rust.py

```diff
@@ -808,6 +814,41 @@ async def test_basic_delivery_with_vapid(
     assert result["headers"]["encryption"] == clean_header
     assert result["data"] == base64url_encode(uuid_data)
     assert result["messageType"] == "notification"
+    # The key we used should not have been registered, so no tracking should
+    # be occurring.
+    log.debug(f"🔍 Reliability: {result.get("reliability_id")}")
+    assert result.get("reliability_id") is None
+
+
+async def test_basic_delivery_with_tracked_vapid(
+    registered_test_client: AsyncPushTestClient,
+    vapid_payload: dict[str, int | str],
+) -> None:
+    """Test delivery of a basic push message with a VAPID header."""
+    uuid_data: str = str(uuid.uuid4())
+    vapid_info = _get_vapid(key=TRACKING_KEY, payload=vapid_payload)
+    # quick sanity check to ensure that the keys match.
+    # (ideally, this should dump as x962, but DER is good enough.)
+    key = cast(
+        ecdsa.VerifyingKey, cast(ecdsa.SigningKey, vapid_info["key"]).get_verifying_key()
+    ).to_der()
+
+    assert key == TRACKING_PUB_KEY.to_der()
+
+    # let's do an offline submit so we can validate the reliability_id survives storage.
+    await registered_test_client.disconnect()
+    await registered_test_client.send_notification(data=uuid_data, vapid=vapid_info)
+    await registered_test_client.connect()
+    await registered_test_client.hello()
+    result = await registered_test_client.get_notification()
+
+    # the following presumes that only `salt` is padded.
+    clean_header = registered_test_client._crypto_key.replace('"', "").rstrip("=")
+    assert result["headers"]["encryption"] == clean_header
+    assert result["data"] == base64url_encode(uuid_data)
+    assert result["messageType"] == "notification"
```

Comment:
We have an enum of `ClientMessageType` in the [async_push_test_client.py](https://github.com/mozilla-services/autopush-rs/blob/3cc5a565f550d482f83d1fc9534cac12bd4cbde0/tests/integration/async_push_test_client.py#L20) which you could add this as a variant, should you wish. Can alter this to simply be `MessageType` as well, since the "client" does double duty in these tests and isn't strictly the client...